pkg/webhooks/workload_webhook.go

```diff
@@ -372,9 +368,20 @@ func validateImmutablePodSets(new, old []kueue.PodSet, path *field.Path) field.E
 	return allErrs
 }
 
-func validateClusterNameUpdate(newObj, oldObj *kueue.Workload, dispatcherName string, statusPath *field.Path) field.ErrorList {
+func isElasticJobReplacement(newObj *kueue.Workload) bool {
+	if !features.Enabled(features.ElasticJobsViaWorkloadSlices) {
+		return false
+	}
+	return workloadslicing.ReplacementForKey(newObj) != nil
+}
+
+func validateClusterNameUpdate(newObj, oldObj *kueue.Workload, statusPath *field.Path) field.ErrorList {
 	var allErrs field.ErrorList
-	if oldObj.Status.ClusterName == nil && newObj.Status.ClusterName != nil && dispatcherName != configapi.MultiKueueDispatcherModeAllAtOnce {
+	if isElasticJobReplacement(newObj) {
```

Comment:
Nit: related to above comment, we could avoid a single-statement (and single-use) function altogether:
```suggestion
		if features.Enabled(features.ElasticJobsViaWorkloadSlices) && workloadslicing.ReplacementForKey(newObj) != nil {
		// Allow setting clusterName when the workload is a valid elastic job replacement.
		return allErrs
	}
```